b2luigi/batch/processes/gbasf2.py

```diff
@@ -326,31 +336,44 @@ def _reschedule_failed_jobs(self):
         Tries to reschedule failed jobs in the project if ``self.max_retries`` has not been reached
         and returns ``True`` if rescheduling has been successful.
         """
+        jobs_to_be_rescheduled = []
+        jobs_hitting_max_n_retries = []
         job_status_dict = get_gbasf2_project_job_status_dict(
             self.gbasf2_project_name, dirac_user=self.dirac_user)
+
         for job_id, job_info in job_status_dict.items():
-            if job_info["Status"] == "Failed":
-                if self.n_retries_by_job[job_id] >= self.max_retries:
-                    warnings.warn(
-                        f"Reached maximum number of rescheduling tries ({self.max_retries}) for job {job_id}.",
-                        RuntimeWarning
-                    )
-                    return False
-                self._reschedule_job(job_id)
-                self.n_retries_by_job[job_id] += 1
-                with open(self.retries_file_path, "w") as retries_file:
-                    json.dump(self.n_retries_by_job, retries_file)
+            if job_info["Status"] == "Failed" or (job_info["Status"] == "Done" and job_info["ApplicationStatus"] != "Done"):
+                if self.n_retries_by_job[job_id] < self.max_retries:
+                    self.n_retries_by_job[job_id] += 1
+                    jobs_to_be_rescheduled.append(job_id)
+                else:
+                    jobs_hitting_max_n_retries.append(job_id)
+
+        if len(jobs_to_be_rescheduled) > 0:
+            self._reschedule_jobs(jobs_to_be_rescheduled)
+            with open(self.retries_file_path, "w") as retries_file:
+                json.dump(self.n_retries_by_job, retries_file)
+
+        if len(jobs_hitting_max_n_retries) > 0:
+            warnings.warn(
+               f"Reached maximum number of rescheduling tries ({self.max_retries}) for following jobs:" +
+               "\n\t".join([str(j) for j in jobs_hitting_max_n_retries]) + "\n",
```

Comment:
You can drop the square brackets for a list comprehension in a function that takes a single argument:
```suggestion
               "\n\t".join(str(j) for j in jobs_hitting_max_n_retries) + "\n",
```
I made sure this works in a ipython terminal:
``` python
In [4]: mylist = list(range(10))
In [5]: "\n".join(str(i) for i in mylist)
Out[5]: '0\n1\n2\n3\n4\n5\n6\n7\n8\n9'
```

Reply:
fixed